docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,476 @@
+validator on startup (when starting with an empty chain).
+
+The file contains a protobuf-encoded list of batches:
+
+.. code-block:: protobuf
+        :caption: File: sawtooth-core/protos/genesis.proto
+
+        message GenesisData {
+            repeated Batch batches = 1;
+        }
+
+The tool should take multiple input batch collections, and combine them
+together into the single list of batches contained in GenesisData. This allows
+independent tools or transaction families to include their own batches, without
+needing to know anything about the genesis process.
+
+The first implementation assumes that the order of the input batches have
+implied dependencies, with each batch being implicitly dependent on the
+previous.  Any dependencies should be verified when the final set of batches is
+produced.  This would be enforced by the use of strict ordering of the batches
+during execution time.  Future implementations may provide a way to verify
+dependencies across input batches.
+
+Transaction family authors who need to provide batches that will be included,
+need to provide their own tool to produce GenesisData, with the batches they
+require for the process. Each individual tool may manage their batch and
+transaction dependencies explicitly within the context of their specific
+genesis batches.
+
+Example
+~~~~~~~
+
+The following example configures the validator to use PoET consensus, the
+Marketplace transaction family and configures the appropriate settings:
+
+.. code-block:: bash
+
+        mktplace genesis \
+          -k <signing-key-file>
+          -o mktplace.batch \
+          --root-participant-name=x \
+          --base-validation-reward=x
+        sawtooth config proposal create \
+          -k <signing-key-file>
+          -o sawtooth-config.batch \
+          sawtooth.transaction.families=mktplace \
+          sawtooth.consensus.algorithm=poet \
+          poet.initial_wait_timer=x \
+          poet.target_wait_time=x \
+          poet.minimum_wait_time=x \
+          poet.certificate_sample_length=x \
+        sawtooth admin genesis \
+          sawtooth-config.batch \
+          mktplace.batch
+
+A genesis.batch file will written to the validator's data directory.
+
+Block Creation
+--------------
+
+On startup, the validator would use the resulting genesis.batch file to produce
+a genesis block under the following conditions:
+
+* The genesis.batch file exists
+* There is no block specified as the chain head
+* There is no ledgerURL specified
+
+If any of these conditions are not met, the validator halts operation.
+
+The validator will load the batches from the file into the pending queue.  It
+will then produce the genesis block through the standard process with the
+following modifications.
+
+First, the execution of the batches will be strictly in the order they have
+been provided.  The Executor will not attempt to reorder them, or drop failed
+transactions.
```

Comment:
There is a sentence for this at the end, but I will move it to this paragraph and reword it.